### PR TITLE
[IMP] tools: Refactor load method to handle non-truthy values consist…

### PR DESCRIPTION
**Pull request by JayshreeMahajan**
Refactor load method to handle non-truthy values consistently

SE-668

### CODE REVIEW

**Review by rajeeshknambiar — APPROVED** [review]

**Review by PeterAlabaster — APPROVED** [review]

File: odoo/tools/config.py [hunk 1]
```diff
@@ -599,9 +599,9 @@ def load(self):
             p.read([self.rcfile])
             for (name,value) in p.items('options'):
                 name = outdated_options_map.get(name, name)
-                if value=='True' or value=='true':
+                if value in ('True', 'true', 'T', 't', 1, '1'):
```
Comment (ws-mps) [inline odoo/tools/config.py]: I think we should consider TRUE and FALSE too.
We can simplify the conditions by converting value to string and then using `lower`.

Comment (PeterAlabaster) [reply odoo/tools/config.py]: something like this?
```suggestion
                if type(value) == str and value.lower in ("t", "true") or value in (1, "1"):
```

Comment (PeterAlabaster) [reply odoo/tools/config.py]: or cast to string beforehand, then just check in `("t", "true", "1")`

**Review by adamrpatrick — APPROVED** [review]

File: odoo/tools/config.py (hunk 1 above)
Comment (kdwyer) [reply odoo/tools/config.py]: Late to the party, but there is also `odoo.tools.misc.str2bool` https://github.com/unipartdigital/odoo/blob/9da977c16237b3bd21638da9aa8a5fd39bcd40ca/odoo/tools/misc.py#L470

**Pull request merged**